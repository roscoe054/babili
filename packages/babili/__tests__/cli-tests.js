jest.autoMockOff();

const { execFile } = require("child_process");
const babiliCli = require.resolve("../lib/cli");

function exec(...options) {
  return new Promise((resolve, reject) => {
    execFile(
      "node",
      [`${babiliCli}`].concat(options),
      (err, stdout, stderr) => {
        if (err) {
          return reject(err);
        }
        if (stdout) {
          return resolve(stdout);
        }
        resolve(stderr);
      }
    );
  });
}

describe("Babili CLI", () => {
  it("should throw error if no option is provided", () => {
    return exec().then(output => {
      expect(output).toMatchSnapshot();
    });
  });

  it("should throw on invalid options", () => {
    return exec("--foo").then(output => {
      expect(output).toMatchSnapshot();
    });
  });

  it("should show help for --help", () => {
    return exec("--help").then(output => {
      expect(output).toMatchSnapshot();
    });
  });

  // TODO - fix
  xit("should print to stdout if --stdin is specified", () => {
    let source = "let abcd = 10, bcdsa = 20";
    return exec("--mangle", "--stdin", `${source}`).then(output => {
      expect(output).toMatchSnapshot();
    });
  });
});
