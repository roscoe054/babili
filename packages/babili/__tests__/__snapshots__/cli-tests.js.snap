// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Babili CLI should show help for --help 1`] = `
"Usage: babili index.js [options]

  Options:
    --mangle    Context and scope aware variable renaming

    --simplify  Simplifies code for minification by reducing statements into
                expressions and making expressions uniform where possible

  "
`;

exports[`Babili CLI should throw error if no option is provided 1`] = `
"Provide filenames/dir or pass --stdin as option
"
`;

exports[`Babili CLI should throw on invalid options 1`] = `
"Provide filenames/dir or pass --stdin as option
Invalid Options passed: foo
"
`;
